File: Examples/R/tests/testParallel.R
test_that("Parallel exports are available", {
  classNames <- c("FMTtask", "FMTreplanningtask", "FMTplanningtask", "FMTtaskhandler")
  for (className in classNames) {
    expect_true(exists(className, inherits = TRUE))
  }
})

test_that("Operating area task follows OSI feature", {
  skipIfFMTFeatureMissing("OSI")
  expect_true(exists("FMTopareaschedulertask", inherits = TRUE))
})

File: Examples/Python/tests/testParallel.py
from FMT import Parallel


def testParallelObjectsCanBeConstructed():
    constructors = {
        "FMTtask": Parallel.FMTtask,
        "FMTreplanningtask": Parallel.FMTreplanningtask,
        "FMTplanningtask": Parallel.FMTplanningtask,
        "FMTtaskhandler": Parallel.FMTtaskhandler,
    }
    for name, constructor in constructors.items():
        assert constructor() is not None
